component/main.jsonnet

```
@@ -100,6 +100,87 @@ local canary = kube._Object('monitoring.appuio.io/v1beta1', 'SchedulerCanary', '
   },
 };
 
+local storageCanaries = std.flattenArrays(std.filterMap(
+  function(storageclass) params.slos.storage.canary._sli.volume_plugins[storageclass] != null,
+  function(storageclass)
+    local p = params.slos.storage.canary._sli.volume_plugins_default_params + com.makeMergeable(params.slos.storage.canary._sli.volume_plugins[storageclass]);
+    local manifestName = 'canary-%s' % if storageclass == '' then 'default' else storageclass;
+    [
+      kube.PersistentVolumeClaim(manifestName) {
+        metadata+: {
+          namespace: params.namespace,
+        },
+        spec+: {
+          accessModes: [ p.accessMode ],
+          resources: {
+            requests: {
+              storage: p.size,
+            },
+          },
+          [if storageclass != '' then 'storageClassName']: storageclass,
+        },
+      },
+      kube._Object('monitoring.appuio.io/v1beta1', 'SchedulerCanary', manifestName) {
+        metadata+: {
+          namespace: params.namespace,
+        },
+        spec: {
+          interval: p.interval,
+          maxPodCompletionTimeout: p.maxPodCompletionTimeout,
+          forbidParallelRuns: true,
+          podTemplate: {
+            metadata: {},
+            spec: {
+              affinity: {
+                nodeAffinity: params.canary_node_affinity,
+              },
+              containers: [
+                {
+                  command: [
+                    'sh',
+                    '-c',
+                  ],
+                  args: [
+                    std.join(';', [
+                      'echo test > /testmount/test',
```

Might make sense to read back, although not sure how valuable that would actually be

```suggestion
                      'echo test > /testmount/test',
                      '[ `cat /testmount/test` = "test" ] || exit 1',
```